system-addon/test/unit/content-src/components/ErrorBoundary.test.jsx

```diff
@@ -0,0 +1,103 @@
+import {ErrorBoundary, ErrorBoundaryFallback} from "content-src/components/ErrorBoundary/ErrorBoundary";
+import {FormattedMessage} from "react-intl";
+import React from "react";
+import {shallow} from "enzyme";
+
+describe("<ErrorBoundary>", () => {
+  it("should render its children if componentDidCatch wasn't called", () => {
+    const wrapper = shallow(<ErrorBoundary ><div className="kids" /></ErrorBoundary>);
+
+    assert.lengthOf(wrapper.find(".kids"), 1);
+  });
+
+  it("should render ErrorBoundaryFallback if componentDidCatch called", () => {
+    const wrapper = shallow(<ErrorBoundary />);
+
+    wrapper.instance().componentDidCatch();
+    // since shallow wrappers don't automatically manage lifecycle semantics:
+    wrapper.update();
+
+    assert.lengthOf(wrapper.find(ErrorBoundaryFallback), 1);
+  });
+
+  it("should render the given FallbackComponent if componentDidCatch called", () => {
+    class TestFallback extends React.PureComponent {
+      render() {
+        return <div className="my-fallback">doh!</div>;
+      }
+    }
+
+    const wrapper = shallow(<ErrorBoundary FallbackComponent={TestFallback} />);
+    wrapper.instance().componentDidCatch();
+    // since shallow wrappers don't automatically manage lifecycle semantics:
+    wrapper.update();
+
+    assert.lengthOf(wrapper.find(TestFallback), 1);
+  });
+
+  it("should pass the given className prop to the FallbackComponent", () => {
+    class TestFallback extends React.PureComponent {
+      render() {
+        return <div className={this.props.className}>doh!</div>;
+      }
+    }
+
+    const wrapper = shallow(
+      <ErrorBoundary FallbackComponent={TestFallback} className="sheep" />);
+    wrapper.instance().componentDidCatch();
+    // since shallow wrappers don't automatically manage lifecycle semantics:
+    wrapper.update();
+
+    assert.lengthOf(wrapper.find(".sheep"), 1);
+  });
+});
+
+describe("ErrorBoundaryFallback", () => {
+  it("should render a <div> with a class of as-error-fallback", () => {
+    const wrapper = shallow(<ErrorBoundaryFallback />);
+
+    assert.lengthOf(wrapper.find("div.as-error-fallback"), 1);
+  });
+
+  it("should render a <div> with the props.className and .as-error-fallback", () => {
+    const wrapper = shallow(<ErrorBoundaryFallback className="monkeys" />);
+
+    assert.lengthOf(wrapper.find("div.monkeys.as-error-fallback"), 1);
+  });
+
+  it("should call window.location.reload(true) if .reload-button clicked", () => {
+    const fakeWindow = {location: {reload: sinon.spy()}};
+    const wrapper = shallow(<ErrorBoundaryFallback windowObj={fakeWindow} />);
+
+    wrapper.find(".reload-button").simulate("click");
+
+    assert.calledOnce(fakeWindow.location.reload);
+    assert.calledWithExactly(fakeWindow.location.reload, true);
```

Comment:
can you do the same trick here? i.e. could these two lines be `assert.calledOnce(fakeWindow.location.reload.withArgs(true));` instead?

Reply:
I would expect that to work, but to me, that feels harder to read.  Maybe it's just because I'm stockholm-syndromed into the old way.  But I think I'll leave it as is.